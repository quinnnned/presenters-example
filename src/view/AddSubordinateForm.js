import React from 'react'
import * as UI from 'react-materialize'

export const handleSubmit = (employee) => (event) => { 
    event.preventDefault()
    const field = event.target[0]
    employee.addSubordinate(field.value)
    field.value = ''
}

export default function AddSubordinateForm(props) {
    const {employee} = props
    return (
        <UI.Row>
            <form 
                onSubmit={handleSubmit(employee)}
                >
                <UI.Input 
                    label="Subordinate Name"
                    required 
                    s={12} 
                    />
            </form>
        </UI.Row>
    )
}
