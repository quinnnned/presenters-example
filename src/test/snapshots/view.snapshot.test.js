import React from 'react'
import View from '../../view'
import AddSubordinateForm from '../../view/AddSubordinateForm'
import renderer from 'react-test-renderer'
import {exampleEmployee} from '../helper'

//mock jquery
global.$ = () => ({ collapsible: () => {} })

describe('View', () => {
    it('renders without crashing', () => {
        expect(
            renderer.create(
                <View person={exampleEmployee} />
            ).toJSON()
        ).toMatchSnapshot();
    })
})

describe('AddSubordinateForm', () => {
    it('renders without crashing', () => {
        expect(
            renderer.create(
                <AddSubordinateForm person={exampleEmployee} />
            ).toJSON()
        ).toMatchSnapshot();
    })
})