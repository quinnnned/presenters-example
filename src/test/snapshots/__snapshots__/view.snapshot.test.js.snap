// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSubordinateForm renders without crashing 1`] = `
<div
  className="row"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="col input-field s12"
    >
      <input
        className=""
        defaultValue={undefined}
        id="input_3"
        onChange={[Function]}
        placeholder={undefined}
        required={true}
        type="text"
      />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_3"
      >
        Subordinate Name
      </label>
    </div>
  </form>
</div>
`;

exports[`View renders without crashing 1`] = `
<div
  className="container"
>
  <h1
    className="center-align"
  >
    Organization Explorer
  </h1>
  <ul
    className="collapsible popout"
    data-collapsible="expandable"
  >
    <li
      className="active"
    >
      <div
        className="collapsible-header active"
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          person
        </i>
        <a
          onClick={[Function]}
        >
          Robert Robertson
        </a>
      </div>
      <div
        className="collapsible-body"
      >
        <div
          className="row"
        >
          <h6>
            Supervisor: 
            <a
              onClick={[Function]}
            >
              Bossy Bosserson
            </a>
          </h6>
          <br />
          <h6>
            Subordinates (
            2
            ):
          </h6>
          <ul
            className="collapsible popout"
            data-collapsible="accordion"
          >
            <li
              className=""
            >
              <div
                className="collapsible-header"
                onClick={[Function]}
              >
                <i
                  className="material-icons"
                >
                  person
                </i>
                <span>
                  <a
                    onClick={[Function]}
                  >
                    John Johnson
                  </a>
                  <a
                    onClick={[Function]}
                  >
                    <i
                      className="material-icons right red-text"
                    >
                      whatshot
                    </i>
                  </a>
                </span>
              </div>
              <div
                className="collapsible-body"
              >
                <div
                  className="row"
                >
                  <h6>
                    Supervisor: 
                    <a
                      onClick={[Function]}
                    >
                      Robert Robertson
                    </a>
                  </h6>
                  <br />
                  <h6>
                    Subordinates (
                    0
                    ):
                  </h6>
                  <ul
                    className="collapsible popout"
                    data-collapsible="accordion"
                  >
                    <li
                      className=""
                    >
                      <div
                        className="collapsible-header"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons"
                        >
                          person_add
                        </i>
                        Add New Subordinate
                      </div>
                      <div
                        className="collapsible-body"
                      >
                        <div
                          className="row"
                        >
                          <form
                            onSubmit={[Function]}
                          >
                            <div
                              className="col input-field s12"
                            >
                              <input
                                className=""
                                defaultValue={undefined}
                                id="input_0"
                                onChange={[Function]}
                                placeholder={undefined}
                                required={true}
                                type="text"
                              />
                              <label
                                className=""
                                data-error={undefined}
                                data-success={undefined}
                                htmlFor="input_0"
                              >
                                Subordinate Name
                              </label>
                            </div>
                          </form>
                        </div>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li
              className=""
            >
              <div
                className="collapsible-header"
                onClick={[Function]}
              >
                <i
                  className="material-icons"
                >
                  person
                </i>
                <span>
                  <a
                    onClick={[Function]}
                  >
                    Bobby Bobberson
                  </a>
                  <a
                    onClick={[Function]}
                  >
                    <i
                      className="material-icons right red-text"
                    >
                      whatshot
                    </i>
                  </a>
                </span>
              </div>
              <div
                className="collapsible-body"
              >
                <div
                  className="row"
                >
                  <h6>
                    Supervisor: 
                    <a
                      onClick={[Function]}
                    >
                      Robert Robertson
                    </a>
                  </h6>
                  <br />
                  <h6>
                    Subordinates (
                    0
                    ):
                  </h6>
                  <ul
                    className="collapsible popout"
                    data-collapsible="accordion"
                  >
                    <li
                      className=""
                    >
                      <div
                        className="collapsible-header"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons"
                        >
                          person_add
                        </i>
                        Add New Subordinate
                      </div>
                      <div
                        className="collapsible-body"
                      >
                        <div
                          className="row"
                        >
                          <form
                            onSubmit={[Function]}
                          >
                            <div
                              className="col input-field s12"
                            >
                              <input
                                className=""
                                defaultValue={undefined}
                                id="input_1"
                                onChange={[Function]}
                                placeholder={undefined}
                                required={true}
                                type="text"
                              />
                              <label
                                className=""
                                data-error={undefined}
                                data-success={undefined}
                                htmlFor="input_1"
                              >
                                Subordinate Name
                              </label>
                            </div>
                          </form>
                        </div>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li
              className=""
            >
              <div
                className="collapsible-header"
                onClick={[Function]}
              >
                <i
                  className="material-icons"
                >
                  person_add
                </i>
                Add New Subordinate
              </div>
              <div
                className="collapsible-body"
              >
                <div
                  className="row"
                >
                  <form
                    onSubmit={[Function]}
                  >
                    <div
                      className="col input-field s12"
                    >
                      <input
                        className=""
                        defaultValue={undefined}
                        id="input_2"
                        onChange={[Function]}
                        placeholder={undefined}
                        required={true}
                        type="text"
                      />
                      <label
                        className=""
                        data-error={undefined}
                        data-success={undefined}
                        htmlFor="input_2"
                      >
                        Subordinate Name
                      </label>
                    </div>
                  </form>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
